packages/playwright-core/src/utils/third_party/tar/headers.js

```diff
@@ -0,0 +1,225 @@
+const ZERO_OFFSET = '0'.charCodeAt(0)
+const USTAR_MAGIC = Buffer.from([0x75, 0x73, 0x74, 0x61, 0x72, 0x00]) // ustar\x00
+const GNU_MAGIC = Buffer.from([0x75, 0x73, 0x74, 0x61, 0x72, 0x20]) // ustar\x20
+const GNU_VER = Buffer.from([0x20, 0x00])
+const MAGIC_OFFSET = 257
+const VERSION_OFFSET = 263
+
+exports.decodeLongPath = function decodeLongPath (buf, encoding) {
+  return decodeStr(buf, 0, buf.length, encoding)
+}
+
+exports.encodePax = function encodePax (opts) { // TODO: encode more stuff in pax
+  let result = ''
+  if (opts.name) result += addLength(' path=' + opts.name + '\n')
+  if (opts.linkname) result += addLength(' linkpath=' + opts.linkname + '\n')
+  const pax = opts.pax
+  if (pax) {
+    for (const key in pax) {
+      result += addLength(' ' + key + '=' + pax[key] + '\n')
+    }
+  }
+  return Buffer.from(result)
+}
+
+exports.decodePax = function decodePax (buf) {
+  const result = {}
+
+  while (buf.length) {
+    let i = 0
+    while (i < buf.length && buf[i] !== 32) i++
+    const len = parseInt(buf.toString('ascii', 0, i), 10)
+    if (!len) return result
+
+    const b = buf.subarray('ascii', i + 1, len - 1)
```

Comment:
Not sure if it's important but len-1 maybe out of the buffer bounds, we may want to throw in that case.